src/Functions/FunctionStrncmp.cpp

```diff
@@ -0,0 +1,445 @@
+
+        if (!WhichDataType(args[3].type).isNativeUInt() || !args[3].column->isConst())
+            throw Exception(
+                ErrorCodes::BAD_ARGUMENTS,
+                "The fourth argument of function {} must be a constant non-negative, but got {}.",
+                getName(),
+                args[3].column->getName());
+
+        if (!WhichDataType(args[4].type).isNativeUInt() || !args[4].column->isConst())
+            throw Exception(
+                ErrorCodes::BAD_ARGUMENTS,
+                "The fifth argument of function {} must be a constant non-negative, but got {}.",
+                getName(),
+                args[4].column->getName());
+    }
+
+    template <bool positive>
+    static Int8 normalComparison(const char * first_str, size_t first_str_len, const char * second_str, size_t second_str_len)
+    {
+        if constexpr (positive)
+            return static_cast<Int8>(memcmpSmallLikeZeroPaddedAllowOverflow15(first_str, first_str_len, second_str, second_str_len));
+        else
+            return static_cast<Int8>(memcmpSmallLikeZeroPaddedAllowOverflow15(second_str, second_str_len, first_str, first_str_len));
+    }
+
+    template <bool positive>
+    static bool isOverflowComparison(size_t first_str_len, size_t first_offset, size_t second_str_len, size_t second_offset, Int8 & result)
+    {
+        if (first_offset >= first_str_len && second_offset >= second_str_len)
+        {
+            result = 0;
+            return true;
+        }
+        else if (first_offset >= first_str_len)
+        {
+            if constexpr (positive)
+                result = -1;
+            else
+                result = 1;
+            return true;
+        }
+        else if (second_offset >= second_str_len)
+        {
+            if constexpr (positive)
+                result = 1;
+            else
+                result = -1;
+            return true;
+        }
+        return false;
+    }
+
+    template <bool positive>
+    static bool isOverflowComparison(
+        size_t first_str_len, size_t first_offset, size_t second_str_len, size_t scond_offset, PaddedPODArray<Int8> & result, size_t rows)
+    {
+        bool is_overflow = false;
+        Int8 res = 0;
+        if (first_offset >= first_str_len && scond_offset >= second_str_len)
+        {
+            res = 0;
+            is_overflow = true;
+        }
+        else if (first_offset >= first_str_len)
+        {
+            if constexpr (positive)
+                res = -1;
+            else
+                res = 1;
+            is_overflow = true;
+        }
+        else if (scond_offset >= second_str_len)
+        {
+            if constexpr (positive)
+                res = 1;
+            else
+                res = -1;
+            is_overflow = true;
+        }
+        for (size_t i = 0; i < rows; ++i)
+            result[i] = res;
+        return is_overflow;
+    }
+
+    template <bool positive = true>
+    void executeStringWithConstString(
```

Comment:
In other places in the code base, such implementation functions would be called `executeStringConst`, in l. 271, it would be `executeFixedStringConst` etc.